RockLib.Configuration.ObjectFactory/ConfigReloadingProxyFactory.cs

```diff
@@ -138,6 +138,8 @@ public static class ConfigReloadingProxyFactory
             throw new ArgumentNullException(nameof(interfaceType));
          if (!interfaceType.IsInterface)
             throw new ArgumentException($"Specified type is not an interface: '{interfaceType.FullName}'.", nameof(interfaceType));
+         if (!typeof(IDisposable).IsAssignableFrom(interfaceType))
+            throw new ArgumentException("The Specified type does not implement IDisposable.");
```

Comment:
Would it be easier for consumers to understand the problem if the error message were something more like:

`$"The specified type '{ interfaceType }' must implement {nameof(IDisposable}"`